### PR TITLE
Implement support for notifications

### PR DESCRIPTION
**Pull request by NickM-27**
This adds support for WebPush notifications from Frigate directly to the browser / PWA. This will enable notifications that can be delivered directly to the user without requiring a separate service and can directly link into the part of Frigate that is relevant to the notification.

Closes https://github.com/blakeblackshear/frigate/issues/7814

To-Do:

- [x] Figure out where to store VAPID (non sensitive)
- [x] Add documentation for enabling notifications and notification requirements
- [x] Try to implement image for chrome along with icon
- [x] Refine the UI and UX
- [x] More testing

### CODE REVIEW

**Review by hawkeye217 — APPROVED** [review]

File: frigate/api/notification.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    key = Vapid01.from_file(os.path.join(CONFIG_DIR, "notifications.pem"))
+    raw_pub = key.public_key.public_bytes(
+        serialization.Encoding.X962, serialization.PublicFormat.UncompressedPoint
+    )
+    return jsonify(utils.b64urlencode(raw_pub)), 200
+
+
+@NotificationBp.route("/notifications/register", methods=["POST"])
+def register_notifications():
+    username = request.headers.get("remote-user", type=str) or "admin"
```
Comment (blakeblackshear) [inline frigate/api/notification.py:42]: I'm not sure this will work when using proxy auth and passing a username from the upstream proxy.

Comment (NickM-27) [reply frigate/api/notification.py]: For now we will treat proxy users as admin, in the future it can be made separate

**Review by blakeblackshear — APPROVED** [review]

**Comment by EduardoLaranjo** [comment]
hi @NickM-27 is this been release?

**Comment by NickM-27** [comment]
No, only in dev builds

**Comment by EduardoLaranjo** [comment]
when do you think you are gonna release this feature ? Thought it would be release in 14.1

**Comment by EduardoLaranjo** [comment]
hi @NickM-27 do you think can you release 0.15 as beta, I happy to be the guinea pig.

**Comment by NickM-27** [comment]
Like I already said there are dev builds you can use, we are not ready for a beta yet

**Pull request merged**